source/index.ts

```diff
@@ -165,9 +165,10 @@ class Conf<T extends Record<string, any> = Record<string, unknown>> implements I
 	@param key - The key of the item to get.
 	@param defaultValue - The default value if the item does not exist.
 	*/
-	get<Key extends keyof T>(key: Key | string): T[Key] | undefined;
-	get<Key extends keyof T, Default = T[Key]>(key: Key | string, defaultValue: Default): T[Key] | Default;
-	get<Key extends keyof T, Default = T[Key]>(key: Key | string, defaultValue?: Default): Default | undefined {
+	get<Key extends keyof T>(key: Key): T[Key];
+	get<Key extends keyof T>(key: Key, defaultValue: Required<T>[Key]): Required<T>[Key];
+	get<Key extends string, TValue = unknown>(key: Key extends keyof T ? never : Key, defaultValue?: TValue): TValue;
```

Comment:
```suggestion
	get<Key extends string, Value = unknown>(key: Key extends keyof T ? never : Key, defaultValue?: Value): Value;
```

Reply:
Could you add a short code comment about the purpose of this overload? It's not immediately clear.

Reply:
Its purpose is commit name: using incorect type for default value should not fallthrough to overload with unknown type.

If you want to put it in source I'm not sure how to do it elegantly 😄

Reply:
Just write what you wrote above, just in a code comment.